in-progress/13000-tx-availability/design.md

```diff
@@ -0,0 +1,91 @@
+
+To mitigate peers from being targetted by DOS attacks, we can attach the block proposal ID to any requests for transactions. The peers being asked can then verify that it is a known, recent proposal. Additionally, the peer can choose to only accept these requests from peers that it gossiped the block to in the first place and can limit the number of such requests that it is willing to service.
+
+Targeting the propagating peer in this way ensures that the transaction 'should' propagate throughout the network albeit at a potentially slower rate than gossiping. Every level of propagation requires a round trip of request/response. 
+
+We don't wait for each level of propagtion to retrieve the transactions as doing so would delay the block proposal. It is possible (even likely) that a sufficient number of validators already have the transactions and can execute immediately.
+
+### Request/Response directly from proposer
+
+Performing request/response more aggresively from the propagator should further increase the likelihood of transactions being generally available to all peers. However, if this is found to still be insufficient then we propose a method of requesting directly from the proposer. This would only be an option for committee members.
+
+Validators would be required to post their Ethereum public key as part of joining the validator set. The proposer can then optionally setup an auth'd endpoint through which it will serve requests for transactions in the block proposal. The details of this endpoint would be encrypted with each committee member's public key and included in the block proposal. The proposer has complete control over this endpoint enabling them to protect themselves against potential DOS attacks by malicious committee members.
+
+### Make prover coordination urls universal
+
+As a leftover from the previous method of prover coordination, provers have additional configuration allowing them to request transactions from other nodes via http. This can be made a universal option for all nodes and incorporated into the TxRetrieval workflow.
+
+### Increasing prover transaction pool requirements
+
+Provers are assumed to be organisations with access to considerable hardware resources and bandwidth capabilities. It is not unreasonable for them to:
+
+1. Run multiple full nodes in addition to the prover node.
+2. Configure every node with a very large transaction pool size, significantly reducing the transaction eviction rate.
+3. Configure every node to have a high peer count, increasing the likelihood of request/response success.
```

Comment:
We could increase the mesh size to D = 12 to be more aggressive. 
AFAIK the values above have no observable improvements